gcp_variant_transforms/transforms/infer_headers.py

```diff
@@ -39,12 +41,37 @@
 class _InferHeaderFields(beam.DoFn):
   """Infers header fields from `Variant` records.
 
-  Two types of fields are inferred:
+  Up to three types of fields are inferred:
+
+  if `infer_headers` is True:
   - The fields are undefined in the headers.
   - The field definitions provided by the headers does not match the field
     values.
+  if `infer_annotation_types` is True:
+  - Fields containing type information of corresponding annotation Info fields.
   """
 
+  def __init__(
+      self,
+      annotation_fields=None,  # type: Union[None, List[str]]
+      infer_headers=False,  # type: bool
+      infer_annotation_types=False  # type: bool
```

Comment:
I think it's better to just use one variable here (e.g. `annotation_fields_to_infer` or a better name :p) that just contains the list of annotation fields. That way, the API is simpler and you dont need to worry about cases where flag is true, but annotation_fields is empty or vice versa.